catalyst_voices/packages/internal/catalyst_voices_blocs/lib/src/campaign/details/campaign_details_bloc.dart

```diff
@@ -0,0 +1,104 @@
+import 'package:catalyst_voices_blocs/src/campaign/details/campaign_details.dart';
+import 'package:catalyst_voices_models/catalyst_voices_models.dart';
+import 'package:catalyst_voices_view_models/catalyst_voices_view_models.dart';
+import 'package:flutter_bloc/flutter_bloc.dart';
+
+final class CampaignDetailsBloc
+    extends Bloc<CampaignDetailsEvent, CampaignDetailsState> {
+  CampaignDetailsBloc() : super(const CampaignDetailsState()) {
+    on<LoadCampaign>(_onLoadCampaign);
+  }
+
+  Future<void> _onLoadCampaign(
+    LoadCampaign event,
+    Emitter<CampaignDetailsState> emit,
+  ) async {
+    final id = event.id;
+
+    emit(
+      state.copyWith(
+        title: const Optional.empty(),
+        listItems: const [],
+      ),
+    );
+
+    final campaign = await _fetchCampaign(id);
+    final listItems = _mapCampaignToListItems(campaign);
+
+    emit(
+      state.copyWith(
+        title: Optional.of(campaign.name),
+        listItems: listItems,
+      ),
+    );
+  }
+
+  Future<Campaign> _fetchCampaign(String id) async {
```

Comment:
In the long run we will fetch data from repositories, you can see the `ProposalRepository` how I did it.

Reply:
Something like:

1. widget calls bloc
2. bloc gets data from repository
3. bloc processes data in some way if it needs to
4. bloc emits new states
5. widget rebuilds